python/pyarrow/io.pxd

```diff
@@ -25,6 +25,8 @@ from pyarrow.includes.libarrow_io cimport (ReadableFileInterface,
 cdef class Buffer:
     cdef:
         shared_ptr[CBuffer] buffer
+        Py_ssize_t shape[1]
+        Py_ssize_t strides[1]
```

Comment:
They need to be `Py_ssize_t` pointers that are valid as long as the buffer is held, so adding them to the struct seemed like the most reasonable option. The few examples of implementing the buffer protocol with Cython all do the same, FWIW.